src/anndata/_io/specs/methods.py

```diff
@@ -632,7 +632,7 @@ def write_vlen_string_array_zarr(
         filters, dtype = (
             ([VLenUTF8()], object)
             if ad.settings.zarr_write_format == 2
-            else (None, str)
+            else (None, np.dtypes.StringDType())
```

Comment:
so a compatible numpy version is guaranteed when zarr v3 is used?

or are we already relying on a compatible numpy version.

Reply:
Great catch, right, they even have code for this and I have seen it before: https://github.com/zarr-developers/zarr-python/blob/54b3d44b36e1876eab93239bc0de43c3d70e1e92/src/zarr/core/strings.py#L15